docs/advanced/working_with_git.rst

```diff
@@ -0,0 +1,207 @@
+
+   * | **Soln:** Please **understand** `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_. 
+       Checkout the branch the pull request represents, count the **number[N]** 
+       of commits you need to `squash <http://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one>`_ 
+       from the pull requests or ``git log`` on that branch, then 
+       ``git rebase -i HEAD~N``. N (**number of commits before**) is usually 2 
+       if you want to **squash 2 commits**. Change the pick to squash. 
+       Save the configuration. Then force push the new commit 
+       ``git push -f origin <branch>``.
+
+#. | If you have **unwanted commits** attached to the pull request or 
+     **history is broken** then you need to do an **interactive rebase** :
+
+   * | **Soln:** It is better to tell others that you are having such problem 
+       as it needs `rewriting history <http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History>`_. 
+       Please **understand** `rebasing <https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request>`_ 
+       and please see your `logs <http://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History>`_. 
+       The **solution** is an `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_ 
+       **Command**: ``git rebase -i HEAD~N`` (N gives the number of the 
+       revisions up in that branch). Try to pick N carefully count the commits 
+       as maintainers will ask you to give the finalized result in a single 
+       commit.
+
+   * | The **interactive rebase** on that branch will show the commands that 
+       runs on that branch and finally shows on the pull request. Delete the 
+       commits you don't need and keep the ones you need. Squash the needed 
+       commits to a single commit.
+
+   * | **Note:** Branches are technically a **tagging system** to commits that 
+       have a **hash value**, they are **relative**, even the master branch. 
+       It can operate with commands like squash, pick and many others. The 
+       commands are played from **top to bottom** on each commit and finally 
+       shown in the pull request. See the `options <https://help.github.com/articles/about-git-rebase/>`_.
+
+Resolving merge conflicts
+-------------------------
+| Conflicts are essentially two or more commits from different branches which 
+  **overlap** because they have **different content** on the same revision. 
+  Understanding `Merge conflict <https://www.youtube.com/watch?v=zz7NuSCH6II>`_, 
+  you can manually resolve it with git.
+
+| Tools like `kdiff3 <https://www.kde.org/applications/development/kdiff3/>`_ 
+  helps you pick that **content of the commit** which you want to get merged in 
+  the final commit. A `tutorial <https://www.youtube.com/watch?v=-CkqiIPAzgQ>`_ 
+  on using **kdiff3**. Kdiff3 can be configured for all types of version 
+  control ranging from **git, svn or mercurial**.
+
+Installation
```

Comment:
'Tips on Installing kdiff' would be a better title.

I would add a note "We recommend that you refer to the kdiff3 documentation for the latest installation instructions."